api/v1beta2/network_types.go

```diff
@@ -430,6 +439,26 @@ type SubnetSpec struct {
 
 	// Tags is a collection of tags describing the resource.
 	Tags Tags `json:"tags,omitempty"`
+
+	// ZoneType defines a zone type for this subnet.
+	//
+	// The valid values are availability-zone, and local-zone.
+	//
+	// Zone types local-zone is not selected to automatically create
+	// resources like Nat Gateway, Network Load Balancers, control plane or compute nodes.
+	//
+	// When local-zone, the public subnets will be associated with the public route table,
+	// the private subnets use the zone in the region to create route tables re-using
+	// NAT Gateways (preferred from the parent zone, the zone type availability-zone in the region,
+	// or first table available).
+	//
+	// +kubebuilder:validation:Enum=availability-zone;local-zone
+	// +optional
+	ZoneType *ZoneType `json:"zoneType,omitempty"`
+
+	// ParentZoneName defines a parent zone name of the zone that the subnet is created.
+	// +optional
```

Comment:
Is this field only valid if `ZoneType == ZoneTypeLocalZone`?

Reply:
Yes (and Wavelenght in the follow up PRs), this field will automatically filled by reconcile from zone describe API. Do you think we need to create some CEL here or just improve the documentation?